packages/legacy/core/App/screens/ListContacts.tsx

```diff
@@ -31,20 +37,28 @@ const ListContacts: React.FC<ListContactsProps> = ({ navigation }) => {
       marginHorizontal: 16,
     },
   })
-  const { records } = useConnections()
-  const [store] = useStore()
-  const { contactHideList } = useConfiguration()
-  // Filter out mediator agents and hidden contacts when not in dev mode
-  let connections: ConnectionRecord[] = records
-  if (!store.preferences.developerModeEnabled) {
-    connections = records.filter((r) => {
-      return (
-        !r.connectionTypes.includes(ConnectionType.Mediator) &&
-        !contactHideList?.includes((r.theirLabel || r.alias) ?? '') &&
-        r.state === DidExchangeState.Completed
-      )
-    })
-  }
+
+  useEffect(() => {
+    const fetchAndSetConnections = async () => {
+      if (!agent) return
+      let orderedContacts = await fetchContactsByLatestMessage(agent as BifoldAgent)
+
+      // if developer mode is disabled, filter out mediator connections and connections in the hide list
+      if (!store.preferences.developerModeEnabled) {
+        orderedContacts = orderedContacts.filter((r) => {
+          return (
+            !r.connectionTypes.includes(ConnectionType.Mediator) &&
+            !contactHideList?.includes((r.theirLabel || r.alias) ?? '') &&
+            r.state === DidExchangeState.Completed
+          )
+        })
+      }
+
+      setConnections(orderedContacts)
+    }
+
+    fetchAndSetConnections()
```

Comment:
worth adding a `.catch()` just in case you end up with `unresolved Promise` for some reason?